system-parachains/coretime/coretime-polkadot/src/migrations.rs

```diff
@@ -0,0 +1,330 @@
+
+		// Iterate through hardcoded leases and check they're all correctly in state and scheduled
+		// in the workplan.
+		for (i, (para_id, until)) in LEASES.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before the leases.
+			let core_id = i as u16 + 4;
+
+			assert!(leases_vec.contains(&(*until, *para_id)));
+
+			// This is the entry found in Workplan
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// They should all be in the workplan for next region.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// For the leases we had before their lease should extend for an additional 11520
+		// timeslices (64 days).
+		for LeaseRecordItem { task, until } in prev_leases.iter() {
+			log::error!("{task}, {until}");
+			assert!(leases_vec.contains(&(*until + 11520, *task)))
+		}
+
+		// Iterate through hardcoded potential renewals and check they're all correctly in state
+		// and scheduled in the workplan.
+		for (i, (para_id, until)) in POTENTIAL_RENEWALS.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before
+			// the leases.
+			let core_id = i as u16 + 4;
+
+			// This is the entry found in Workplan and PotentialRenewals.
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// Make sure they're not in the leases.
+			assert!(!leases.contains(&LeaseRecordItem { until: *until, task: *para_id }));
+
+			// Ensure they can renew in the next region.
+			assert_eq!(
+				PotentialRenewals::<Runtime>::get(PotentialRenewalId {
+					core: core_id,
+					when: sale_info.region_end + 5040,
+				}),
+				Some(PotentialRenewalRecord {
+					price: 1_000_000_000_000,
+					completion: pallet_broker::CompletionStatus::Complete(workload.clone())
+				})
+			);
+
+			// They should all be in the workplan for next sale.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// Walk the workplan at timeslice 287565 and make sure there is an entry for every 62 cores.
+		for i in 0..61 {
+			// TODO - finalise
+			assert!(Workplan::<Runtime>::get((287565, i)).is_some());
+		}
+
+		// Ensure we have requested the correct number of cores.
+		assert!(frame_system::Pallet::<Runtime>::read_events_no_consensus().any(|e| {
+			match e.event {
+				crate::RuntimeEvent::Broker(
+					pallet_broker::Event::<Runtime>::CoreCountRequested { core_count },
+				) => {
+					log::info!("{core_count:?}");
+
+					core_count == 62
+				},
+				_ => false,
+			}
+		}));
+
+		Ok(())
+	}
+}
+
+// Incorrect potential renewals in state
+const INCORRECT_RENEWAL_IDS: [(u16, u32); 6] =
+	[(6, 292605), (5, 292605), (13, 292605), (15, 292605), (47, 292605), (44, 292605)];
```

Comment:
Checked with chain state: Matching. (This is the entire list of currently registered potential renewals, as expected.)